### PR TITLE
[GraphBolt] Better naming for split key variable.

### PR DESCRIPTION
**Pull request by mfbalin**
## Checklist
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [ ] I've leverage the [tools](https://docs.google.com/document/d/1iHyj7zlmygKSk5gBPsqIqL5ASPzJSPREaNT_QdsiYA4/edit) to beautify the python and c++ code.
- [ ] The PR is complete and small, read the [Google eng practice (CL equals to PR)](https://google.github.io/eng-practices/review/developer/small-cls.html) to understand more about small PR. In DGL, we consider PRs with less than 200 lines of core code change are small (example, test and documentation could be exempted).
- [ ] All changes have test coverage
- [ ] Code is well-documented
- [ ] To the best of my knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change
- [ ] Related issue is referred in this PR
- [ ] If the PR is for a new model/paper, I've updated the example index [here](../examples/README.md).

### CODE REVIEW

**Review by frozenbugs — APPROVED** [review]

File: graphbolt/src/cache_policy.h [hunk 1]
```diff
@@ -134,8 +135,8 @@ struct CacheKey {
   // Access only through an std::atomic_ref instance atomically.
   int8_t reference_count_;
   // Keys are restricted to be 48-bit unsigned integers.
-  uint16_t key_upper_;
-  uint32_t key_lower_;
+  uint16_t key_upper_16_bits_;
```
Comment (frozenbugs) [inline graphbolt/src/cache_policy.h:138]:
```suggestion
  uint16_t key_first_16_bits_;
```

File: graphbolt/src/cache_policy.h [hunk 2]
```diff
@@ -134,8 +135,8 @@ struct CacheKey {
...
+  uint32_t key_lower_32_bits_;
```
Comment (frozenbugs) [inline graphbolt/src/cache_policy.h:139]:
```suggestion
  uint32_t key_last_32_bits_;
```

File: graphbolt/src/cache_policy.h (hunk 1 above)
Comment (mfbalin) [reply graphbolt/src/cache_policy.h:138]: I don't think first and last is more clear than upper and lower. I think the convention is the following:

https://stackoverflow.com/questions/19063210/what-is-lower-and-higher-bits

https://en.wikipedia.org/wiki/Bit_numbering#:~:text=The%20LSb%20is%20sometimes%20referred,bit%20or%20left%2Dmost%20bit.

Comment (mfbalin) [reply graphbolt/src/cache_policy.h:138]: The LSb is sometimes referred to as the low-order bit or right-most bit, due to the convention in [positional notation](https://en.wikipedia.org/wiki/Positional_notation) of writing less significant digits further to the right. The MSb is similarly referred to as the high-order bit or left-most bit.

**Pull request merged**